interbtc-spec/docs/source/spec/reward.rst

```diff
@@ -0,0 +1,329 @@
+
+Invariants
+~~~~~~~~~~
+
+* For each ``currencyId``,
+
+  * ``TotalStake[currencyId]`` MUST be equal to the sum of ``Stake[currencyId, accountId]`` over all accounts.
+  * ``TotalReward[currencyId]`` MUST be equal to the sum of ``Stake[currencyId, accountId] * RewardPerToken[currencyId] - RewardTally[currencyId, accountId]`` over all accounts.
+  * For each ``accountId``,
+  
+    * ``RewardTally[currencyId, accountId]`` MUST be smaller than or equal to ``Stake[currencyId, accountId] * RewardPerToken[currencyId]``
+    *  ``Stake[currencyId, accountId]`` MUST NOT be negative
+    * ``RewardTally[currencyId, accountId]`` MUST NOT be negative
+
+Data Model
+~~~~~~~~~~
+
+Maps
+----
+
+TotalStake
+..........
+
+The total stake deposited to the reward with the given currency.
+
+TotalRewards
+............
+
+The total unclaimed rewards in the given currency distributed to this reward pool. This value is currently only used for testing purposes.
+
+RewardPerToken
+..............
+
+The amount of reward the stakeholders get for the given currency per unit of stake.
+
+Stake
+.....
+
+The stake in the given currency for the given account.
+
+RewardTally
+...........
+
+The amount of rewards in the given currency a given account has already withdrawn, plus a compensation that is added on stake changes.
+
+
+Functions
+~~~~~~~~~
+
+.. _participants:
+
+participants
+------------
+
+This function gets all the participants in the reward pool.
+
+
+Specification
+.............
+
+*Function Signature*
+
+``participants()``
+
+*Postconditions*
+
+* This function MUST return a list of every account id that has, or has had, a stake in any of the reward pools.
+
+
+
+
+.. _getTotalRewards:
+
+getTotalRewards
+---------------
+
+This function gets the total amount of rewards distributed in the pool with the given currencyId.
+
+Specification
+.............
+
+*Function Signature*
+
+``getTotalRewards(currencyId)``
+
+*Parameters*
+
+* ``currencyId``: Determines of which currency the amount is returned. 
+
+*Postconditions*
+
+* The function MUST return the total amount of rewards that have been distributed in the given currency. 
+
+
+
+
+
+
+.. _depositStake:
+
+depositStake
+------------
+
+Adds a stake for the given account and currency in the reward pool.
+
+Specification
+.............
+
+*Function Signature*
+
+``depositStake(currencyId, accountId, amount)``
+
+*Parameters*
+
+* ``currencyId``: The currency for which to add the stake
+* ``accountId``: The account for which to add the stake
+* ``amount``: The amount by which the stake is to increase
+
+*Events*
+
+* ``DepositStake(currencyId, accountId, amount)``
```

Comment:
What I've been doing in the other PRs (to limit duplication) is referencing the event directly.

Reply:
```suggestion
* :ref:`depositStakeEvent`
```